x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDatafeedsStatsAction.java

```diff
@@ -77,6 +79,10 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
         // 4. Grab runtime state
         ActionListener<Map<String, DatafeedTimingStats>> datafeedTimingStatsListener = ActionListener.wrap(timingStatsByJobId -> {
             responseBuilder.setTimingStatsMap(timingStatsByJobId);
+            GetDatafeedRunningStateAction.Request datafeedRunningStateAction = new GetDatafeedRunningStateAction.Request(
```

Comment:
The cost in the runnstate execution is the transport requests out to the relevant nodes. If the parent transport task is cancelled, the child one (this datafeed running state action) will not even be executed and cancelled. 

If it starts to execute and is serialized out to the task nodes, we lose visibility. I think the tasks framework will cancel the individual calls to the nodes if the parent is cancelled (since they are all transport tasks and the parent task is cancellable). 

But, gathering the runtime state (once its on the task node) is so cheap, checking if the task is cancelled before getting the info there is not necessary.